src/metabase/db/custom_migrations.clj

```diff
@@ -188,27 +188,25 @@
                                                            ["field" y {:source-field x}])
                                             ref ref)]
                        k k))]
-    (-> viz-settings
-        (json/parse-string)
-        (m/update-existing "column_settings" update-keys
-                           (fn [k]
-                             (-> k
-                                 (json/parse-string)
-                                 (vec)
-                                 (old-to-new)
-                                 (json/generate-string))))
-        (json/generate-string))))
+    (m/update-existing viz-settings "column_settings" update-keys
+                       (fn [k]
+                         (-> k
+                             json/parse-string
+                             vec
+                             old-to-new
+                             json/generate-string)))))
```

Comment:
The parsing/generating is now done in migrations since migrations of revisions require that parsing and generation is done one level higher than visualization_settings